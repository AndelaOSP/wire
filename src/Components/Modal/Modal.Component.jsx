import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

import './Modal.scss';

class Modal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      position: null,
      location: null
    };
  }

  /**
   * Method to handle email selection
   */
  handleEmailChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  /**
   * Method to handle position selection
   */
  handlePositionChange = (event, index, value) => {
    this.setState({ position: value });
  };

  /**
   * Method to handle location selection
   */
  handleLocationChange = (event, index, value) => {
    this.setState({ location: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    let { email, position, location } = this.state;
    this.props.handleInvite(email, position, location);
  };

  render() {
    const { open } = this.props;
    const styles = {
      marginTop: '3.3rem',
      marginLeft: '.85rem',
      border: '1px solid #e0e0e0',
      width: '92%',
      height: '3.2rem'
    };
    return (
      <div id="myModal" className="modal" style={{ display: open ? 'block' : 'none' }}>
        <div className="modal-content">
          <span className="close" onClick={this.props.handleClose}>
            &times;
          </span>
          <div className="modal-header">
            <p className="modal-heading">Add Users</p>
            <div className="modal-underline" />
          </div>
          <form className="modal-body" onSubmit={this.handleSubmit}>
            <input
              type="email"
              name="email"
              id="email-input"
              placeholder="Email Address"
              value={this.state.email}
              onChange={this.handleEmailChange}
            />
            <SelectField
              underlineStyle={{ display: 'none' }}
              iconStyle={{ display: 'none' }}
              labelStyle={{ top: '-0.3vh', marginLeft: '1rem', color: '#95989a' }}
              hintText="Position"
              hintStyle={{ top: '1.4vh', marginLeft: '1rem', color: '#95989a' }}
              value={this.state.position}
              onChange={this.handlePositionChange}
              style={styles}
            >
              <MenuItem value="Admin" primaryText="Admin" />
              <MenuItem value="Assignee" primaryText="Assignee" />
            </SelectField>
            <SelectField
              underlineStyle={{ display: 'none' }}
              iconStyle={{ display: 'none' }}
              labelStyle={{ top: '-0.3vh', marginLeft: '1rem', color: '#95989a' }}
              hintText="Country"
              hintStyle={{ top: '1.4vh', marginLeft: '1rem', color: '#95989a' }}
              value={this.state.location}
              onChange={this.handleLocationChange}
              style={styles}
            >
              <MenuItem value="Kenya" primaryText="Kenya" />
              <MenuItem value="Nigeria" primaryText="Nigeria" />
              <MenuItem value="USA" primaryText="USA" />
              <MenuItem value="Uganda" primaryText="Uganda" />
            </SelectField>
            <div className="invite-button">
              <button type="submit" id="send-invite" onClick={this.props.handleClose}>
                Send Invite
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func,
  handleInvite: PropTypes.func
};

export default Modal;
