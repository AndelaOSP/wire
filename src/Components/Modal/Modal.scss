body {font-family: Arial, Helvetica, sans-serif;}

/* The Modal (background) */
.modal {
    display: none; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%; /* Full width */
    height: 100%; /* Full height */
    overflow: auto; /* Enable scroll if needed */
    background-color: rgb(0,0,0); 
    background-color: rgba(0,0,0,0.4); 
}

/* Modal Content */
.modal-content {
    background-color: #fefefe;
    margin: auto;
    padding: 20px;
    border: 1px solid #888;
    width: 20%;

    .modal-heading{
        font-size: 30px;
 
    }
    
    .modal-body{

        display: flex;
        flex-direction: column;

        #email-input{

            height: 2.5rem;
            margin-top: 3rem;
            padding-left: 1rem;

        }
         #position-input{

            height: 2.5rem;
            margin-top: 3rem;
            padding-left: 1rem;
        }
        
        .invite-button{
            text-align: center;
        }

        #send-invite{

            height: 2.7rem;
            margin-top: 5rem;
            background-color: #3359df;
            border-radius: 5px;
            width: 15rem;
            color: white;
            font-family: 'DIN Pro Light';
            font-size: 20px; 
        }
    }
}

/* The Close Button */
.close {
    color: #aaaaaa;
    float: right;
    font-size: 28px;
    font-weight: bold;
}

.close:hover,
.close:focus {
    color: #000;
    text-decoration: none;
    cursor: pointer;
}

