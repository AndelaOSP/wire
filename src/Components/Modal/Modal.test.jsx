import React from 'react';
import { shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Modal from './Modal.Component';

const props = {
  handleInvite: jest.fn(),
  handleClose: jest.fn(),
  open: false,
};

describe('Modal component', () => {
  it('should render without crashing', () => {
    const modal = shallow(<Modal />);
    expect(shallowToJSON(modal)).toMatchSnapshot();
  });

  it('should handle form submit', () => {
    const modal = shallow(<Modal {...props} />);
    const form = modal.find('.modal-body');
    expect(form.length).toEqual(1);
    const spy = jest.spyOn(modal.instance().props, 'handleInvite');
    form.simulate('submit', { preventDefault: jest.fn() });
    expect(spy).toBeCalled();
  });

  it('should handle email change', () => {
    const modal = shallow(<Modal {...props} />);
    const emailInput = modal.find('#email-input');
    expect(modal.state().email).toEqual('');
    expect(emailInput.length).toEqual(1);
    emailInput.simulate('change', {
      preventDefault: jest.fn,
      target: { value: 'user@user.com', name: 'email' },
    });
    expect(modal.state().email).toEqual('user@user.com');
  });

  it('should handle location change', () => {
    const modal = shallow(<Modal {...props} />);
    const selectField = modal.find('#location-select-field');
    expect(selectField.length).toEqual(1);
    selectField.simulate('change');
    selectField.props().onChange({}, 1, 'Nigeria');
    expect(modal.state().location).toEqual('Nigeria');
  });

  it('should handle position change', () => {
    const modal = shallow(<Modal {...props} />);
    const positionField = modal.find('#position-select-field');
    expect(positionField.length).toEqual(1);
    positionField.simulate('change');
    positionField.props().onChange({}, 1, 'new position');
    expect(modal.state().position).toEqual('new position');
  });
});
