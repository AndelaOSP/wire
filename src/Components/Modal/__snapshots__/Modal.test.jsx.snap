// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render without crashing 1`] = `
<div
  className="modal"
  id="myModal"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="modal-content"
  >
    <span
      className="close"
      onClick={[Function]}
    >
      Ã—
    </span>
    <div
      className="modal-header"
    >
      <p
        className="modal-heading"
      >
        Invite User
      </p>
      <div
        className="modal-underline"
      />
    </div>
    <form
      className="modal-body"
      onSubmit={[Function]}
    >
      <input
        id="email-input"
        name="email"
        onChange={[Function]}
        placeholder="Example@andela.com"
        type="email"
        value=""
      />
      <SelectField
        autoWidth={false}
        disabled={false}
        fullWidth={false}
        hintStyle={
          Object {
            "color": "#95989a",
            "marginLeft": "1rem",
            "top": "1.4vh",
          }
        }
        hintText="Select Position"
        iconStyle={
          Object {
            "display": "none",
          }
        }
        id="position-select-field"
        labelStyle={
          Object {
            "color": "#95989a",
            "marginLeft": "1rem",
            "top": "-0.3vh",
          }
        }
        multiple={false}
        onChange={[Function]}
        style={
          Object {
            "border": "1px solid #e0e0e0",
            "borderRadius": "5px",
            "height": "3.2rem",
            "marginLeft": ".85rem",
            "marginTop": "3.3rem",
            "width": "92%",
          }
        }
        underlineStyle={
          Object {
            "display": "none",
          }
        }
        value={null}
      >
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Admin"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="Admin"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Assignee"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="Assignee"
        />
      </SelectField>
      <SelectField
        autoWidth={false}
        disabled={false}
        fullWidth={false}
        hintStyle={
          Object {
            "color": "#95989a",
            "marginLeft": "1rem",
            "top": "1.4vh",
          }
        }
        hintText="Select Country"
        iconStyle={
          Object {
            "display": "none",
          }
        }
        id="location-select-field"
        labelStyle={
          Object {
            "color": "#95989a",
            "marginLeft": "1rem",
            "top": "-0.3vh",
          }
        }
        multiple={false}
        onChange={[Function]}
        style={
          Object {
            "border": "1px solid #e0e0e0",
            "borderRadius": "5px",
            "height": "3.2rem",
            "marginLeft": ".85rem",
            "marginTop": "3.3rem",
            "width": "92%",
          }
        }
        underlineStyle={
          Object {
            "display": "none",
          }
        }
        value={null}
      >
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Kenya"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="Kenya"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Nigeria"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="Nigeria"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="USA"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="USA"
        />
        <MenuItem
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          primaryText="Uganda"
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          value="Uganda"
        />
      </SelectField>
      <div
        className="invite-button"
      >
        <button
          id="send-invite"
          onClick={[Function]}
          type="submit"
        >
          Send Invite
        </button>
      </div>
    </form>
  </div>
</div>
`;
